Add aliased command to which output

Gymea opened the pull request:
This is for issue #1713.

It seems that the others working on this issue dropped it. 

Old behaviour:
```
> alias s = { git status -sb }
> which s
───┬─────┬───────────────┬─────────
 # │ arg │     path      │ builtin
───┼─────┼───────────────┼─────────
 0 │ s   │ Nushell alias │ No
───┴─────┴───────────────┴─────────
```

New behaviour:
```
> alias s = { git status -sb }
> which s
───┬─────┬───────────────────────────────────┬─────────
 # │ arg │               path                │ builtin
───┼─────┼───────────────────────────────────┼─────────
 0 │ s   │ Nushell alias: { git status -sb } │ No
───┴─────┴───────────────────────────────────┴─────────
```

This is my first PR, so apologies if I've done something incorrect.

LhKipp commented:
Good work. Seems like some of the tests are failing. You can run the tests with the following command:
```shell
cargo test --package nu-cli -- commands::which
```

You can also check out CONTRIBUTING.md for more helpfull commands.

Gymea commented:
I've updated the tests that were failing now that the expected output is different, and fixed an issue with the first commit where parameters weren't being displayed.

jonathandturner commented:
Nice! That'll be really helpful

deoradh commented:
.. I was just thinking this could be a good thing to have.  Thanks!
